pynest/nest/lib/hl_api_spatial.py

```diff
@@ -1129,6 +1289,150 @@ def PlotTargets(src_nrn, tgt_layer, syn_type=None, fig=None,
+    fig : [None | matplotlib.figure.Figure object], optional, default: None
+        Matplotlib figure to plot to. If not given, a new figure is created.
+    mask : [None | dict], optional, default: None
+        Draw mask with targets; see :py:func:`.PlotProbabilityParameter` for details.
+    probability_parameter : [None | Parameter], optional, default: None
+        Draw connection probability with targets; see :py:func:`.PlotProbabilityParameter` for details.
+    tgt_color : [None | any matplotlib color], optional, default: 'red'
+        Color used to mark target node position
+    tgt_size : float, optional, default: 50
+        Size of target marker (see scatter for details)
+    src_color : [None | any matplotlib color], optional, default: 'blue'
+        Color used to mark source node positions
+    src_size : float, optional, default: 20
+        Size of source markers (see scatter for details)
+    mask_color : [None | any matplotlib color], optional, default: 'red'
+        Color used for line marking mask
+    probability_cmap : [None | any matplotlib cmap color], optional, default: 'Greens'
+        Color used for lines marking probability parameter.
+
+    Returns
+    -------
+    matplotlib.figure.Figure object
+
+    See also
+    --------
+    PlotTargets: Plot all targets of source neuron in a target layer.
+    GetSourceNodes: Obtain sources of a target in a given source layer.
+    GetSourcePositions: Obtain positions of sources of target in a given source layer.
+    probability_parameter: Add indication of connection probability and mask to axes.
+    PlotLayer: Plot all nodes in a spatially distributed population.
+    matplotlib.pyplot.scatter : matplotlib scatter plot.
+
+    Notes
+    -----
+    * Do **not** use this function in distributed simulations.
+
+    **Example**
+        ::
+
+            import nest
+            import matplotlib.pyplot as plt
+
+            # create a spatial population
+            s_nodes = nest.Create('iaf_psc_alpha', positions=nest.spatial.grid(shape=[11, 11], extent=[11., 11.]))
+
+            # connectivity specifications with a mask
+            conndict = {'rule': 'pairwise_bernoulli', 'p': 1.,
+                        'use_on_source': True,
+                        'mask': {'rectangular': {'lower_left' : [-2.0, -1.0],
+                                                 'upper_right': [2.0, 1.0]}}}
+
+            # connect population s_nodes with itself according to the given
+            # specifications
+            nest.Connect(s_nodes, s_nodes, conndict)
+
+            # plot the targets of a source neuron
+            nest.PlotSources(s_nodes, s_nodes[4])
+            plt.show()
+    """
+
+    # import pyplot here and not at toplevel to avoid preventing users
+    # from changing matplotlib backend after importing nest
+    import matplotlib.pyplot as plt
+
+    if not HAVE_MPL:
+        raise ImportError("Matplotlib could not be imported")
```

Comment:
Should this check be before we import pyplot? And do we even need it? It's just a slightly nicer error message than the standard one.

Reply:
I think we do not necessarily need it, for the exact reason you mentioned.
I do it here since the other plotting functions do it exactly this way as well.
I can remove it, I then would for consistency need to remove all it's appearances in the script.
So maybe it should be another PR? What do you think?

Reply:
As it stands now, if the user doesn't have Matplotlib, an error will be raised at line 1369, making the check of `HAVE_MPL` useless. While I think the added function should be consistent with the other functions, not adding useless code is more important. Therefore I suggest to remove the check here. Fixing the other functions is not within the scope of this PR, so I will create an issue for that.

Reply:
I agree, done in 58b4b98.